### DISCUSSION TITLE
Virtual method table detection correction

### DISCUSSION THREAD

**Discussion by MICRDEV** [Vector35/binaryninja-api#2012] [General] [answered]
Hello!

vtables are not detected automatically in my case. Is there a way to tweak the detection, so I don't need to define each one manually?

Here is a example of vtable which wasn't detected:
![image](https://user-images.githubusercontent.com/25588082/94944094-6b57c900-04e1-11eb-97b4-e17584ef30ce.png)

I'm sure that starting from 0x8d93a8 there are vtable methods pointers.

**Comment by psifertex** [comment]
Can you show a bit more about the code that references that jump table (or even better, attach the binary? put it in a zip if it's not sensitive). These sorts of failures happen when there's a failure of the possible value set to track the ranges the jump table could be.

**Comment by MICRDEV** [comment, marked as answer]
it is referenced in the constructor, like this
![image](https://user-images.githubusercontent.com/25588082/94947707-c213d180-04e6-11eb-9d0e-9a997ea019be.png)

**Reply by psifertex** [reply]
Ahh, sorry, I missed the title and was just thinking "missing jump table" from the previous conversation we had. For C++ classes you'll need to create a structure and apply it at that location. There's no auto-class recovery in Binary Ninja at this time. You might have a look at: 

https://blog.trailofbits.com/2017/02/13/devirtualizing-c-with-binary-ninja/

For the general case of missing jump tables I originally misunderstood you to be asking about, here's a list of the ways that you can use to recover jump tables:

https://binary.ninja/2020/09/10/user-informed-dataflow.html#jump-table-example
